### ISSUE TITLE
I want to use tmux to ensure my processes don't get killed on disconnect

### ISSUE THREAD

**Issue by ericcurtin** [tmux/tmux#2445]
Version information:

```
uname -sp && tmux -V && echo $TERM
Linux x86_64
tmux 3.0a
xterm-256color
```

First of all, I would like to thank everyone for creating such a useful tool. There is a single feature of tmux which is invaluable to me right now. The don't kill child processes when my VPN/ssh connection drops feature (the joys of remote working during COVID). I use xfce4-terminal, openssh client/server, bash, pretty standard stuff.

Previously, logging in with just openssh (no tmux) to a server, my terminal would not be taken over, not be filled with lines of scrollback, not leave my command prompt at the top of the screen (I want to be able to see what I was running before easily). I use my mouse for vim over openssh occasionally, so I would like this to remain completely unaffected.

Long story short, I want tmux to behave as if I was just using openssh alone. What I have done to achieve this so far:

Added the following to end of .bashrc on remote end:

`[ -z "$TMUX"  ] && { tmux attach || exec tmux new-session && exit;}`

to autostart (or attach to) tmux on ssh login.

Added the following to end of .tmux.conf on remote end:

```
set -g terminal-overrides 'xterm*:smcup@:rmcup@'
set -g status off
```

to have mouse behave like it did and to remote status bar at the bottom. I think it's just the scrollback, blank lines behavior I need to change now. I would really appreciate it if someone could help me with that!

**Comment by nicm** [comment]
tmux has its own scrollback and there is no way for it to use the terminal scrollback

**Comment by ericcurtin** [comment]
I understand, but it repositions my terminal as if I called /usr/bin/clear command, this is what I'm trying to avoid.

Maybe that describes it better I hope!

**Comment by ericcurtin** [comment]
It clears on login/logout etc. tmux having it's own buffer is pretty neat! It's all the clearing I want to avoid.

**Comment by ericcurtin** [comment]
Maybe I should rename this as no clear feature request?

**Comment by ericcurtin** [comment]
I see the functionality is called E3 functionality, I don't want tmux to ever call E3, only subprocesses (like vim, etc.)

**Comment by nicm** [comment]
tmux has to know the state of the terminal and there is no way for it to know what was there when it started. It has to start in a known state. The best you can do is leave smcup/rmcup on and then the previous state will be restored when tmux exits.

E3 clears the scrollback, not the terminal. tmux does not send E3 on start or exit.

**Comment by ericcurtin** [comment]
I think I understand, I'd be happy to start at the bottom of the buffer again, even if I lost previous output I wouldn't care.

Is it for the I had vim or some other program opened when I disconnected kind of case?

**Comment by nicm** [comment]
If tmux doesn't know what is on the terminal or what state it is in then it can't correctly draw whatever applications inside tmux send or anything else it wants to draw.

You want tmux to be transparent but that is not how tmux works and there is no way to make it transparent, it is not like ssh which just forwards data between ptys.

**Comment by ericcurtin** [comment]
Completely understand thanks.
